packages/ai/src/image.ts

```diff
@@ -0,0 +1,77 @@
+import { createApi } from "unsplash-js";
+import { traverseTemplateAsync } from "@webstudio-is/jsx-utils";
+import type {
+  EmbedTemplateProp,
+  WsEmbedTemplate,
+} from "@webstudio-is/react-sdk";
+
+const queryImageAndMutateNode = async (
+  api: ReturnType<typeof createApi>,
+  node: WsEmbedTemplate[number]
+) => {
+  if (node.type === "instance" && node.component === "Image") {
+    if (node.props === undefined) {
+      return;
+    }
+    const src = node.props.find((prop) => prop.name === "src");
+    // regenerate images when not specified, empty
+    // or absolute url has wsai=true in search params
+    let canRegenerate = src === undefined;
+    if (src?.type === "string") {
+      canRegenerate = src.value.trim() === "";
+      try {
+        canRegenerate = new URL(src.value).searchParams.get("wsai") === "true";
+      } catch {
+        // empty block
+      }
+    }
+    const alt = node.props.find((prop) => prop.name === "alt");
+    if (canRegenerate === false || alt?.type !== "string") {
+      return;
+    }
+    const photosSearchResult = await api.search.getPhotos({
+      query: alt.value,
+    });
+    if (photosSearchResult.type === "success") {
+      const [result] = photosSearchResult.response.results;
+      const url = new URL(result.urls.raw);
+      url.searchParams.set("wsai", "true");
+      const src: EmbedTemplateProp = {
+        name: "src",
+        type: "string",
+        value: url.href,
+      };
+      const srcIndex = node.props.findIndex((prop) => prop.name === "src");
+      if (srcIndex === -1) {
+        node.props.push(src);
+      } else {
+        node.props[srcIndex] = src;
+      }
+      const hasObjectFit = node.styles?.some(
+        (styleDecl) => styleDecl.property === "objectFit"
+      );
+      if (hasObjectFit === false) {
+        node.styles = node.styles ?? [];
+        node.styles.push({
+          property: "objectFit",
+          value: { type: "keyword", value: "cover" },
+        });
+      }
```

Comment:
hmm, I know we want to do this for an image in general, but I don't know why we are doing it here, where we set an ai generated image.

feels like this should be one and the same place for upload and ai, because this is an optimization for rendering and we shouldn't have 2 places that do that in case we change this

Reply:
It's not optimization. Without this images scaled disproportionally when ai set some sizes.

Reply:
got it, I thought this is the layout shift optimization we also did on the image with upload